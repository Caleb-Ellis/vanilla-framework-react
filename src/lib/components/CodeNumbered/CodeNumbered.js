import React from 'react';
import PropTypes from 'prop-types';
import './CodeNumbered.css';

const CodeNumbered = props => {
  const lines = props.lines.map((lines, i) => {
    return (
      <span key={i} className="code-line">{ props.lines[i] }</span>
    );
  });

  return (
    <pre className="p-code-numbered">
      <code>
        { lines }
      </code>
    </pre>
  );
}

CodeNumbered.propTypes = {
  lines: PropTypes.array.isRequired,
};

export default CodeNumbered;
