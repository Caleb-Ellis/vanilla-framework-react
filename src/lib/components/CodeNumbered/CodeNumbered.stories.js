import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import CodeNumbered from './CodeNumbered';

storiesOf('Code Numbered', module)
  .add('Default',
    withInfo('The code numbered pattern can be used when displaying large blocks of code to enable users to quickly reference a specific line.')(() =>
      <CodeNumbered lines={[
        "this is a code sample line 1",
        "this is another code sample line 2 this is another code sample line 2 this is another code sample line 2",
        "this is the last code sample 3, and everyone knows this is the best code sample"
      ]} />,
    ),
  );
