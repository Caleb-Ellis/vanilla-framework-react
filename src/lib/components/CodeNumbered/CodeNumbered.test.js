import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import CodeNumbered from './CodeNumbered';

describe('CodeNumbered component', () => {
  it('should compare with a snapshot', () => {
    const codenumbered = ReactTestRenderer.create(<CodeNumbered lines={[
      "this is a code sample line 1",
      "this is another code sample line 2 this is another code sample line 2 this is another code sample line 2",
      "this is the last code sample 3, and everyone knows this is the best code sample"
    ]} />);
    const json = codenumbered.toJSON();
    expect(json).toMatchSnapshot();
  });
});
