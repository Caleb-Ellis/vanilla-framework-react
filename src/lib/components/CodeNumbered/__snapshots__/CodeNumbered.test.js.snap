// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeNumbered component should compare with a snapshot 1`] = `
<pre
  className="p-code-numbered"
>
  <code>
    <span
      className="code-line"
    >
      this is a code sample line 1
    </span>
    <span
      className="code-line"
    >
      this is another code sample line 2 this is another code sample line 2 this is another code sample line 2
    </span>
    <span
      className="code-line"
    >
      this is the last code sample 3, and everyone knows this is the best code sample
    </span>
  </code>
</pre>
`;
